### PR TITLE
fix(helmcharts): Allow customization of image tags in Helm chart

### LINKED ISSUE #3988: Allow customization of image tag in Helm chart
### Description

Right now we pull the image tag from app version in the Chart file, however we should allow this to be overridden in the case that someone wants to use a different build, or specify images by digest instead of tag. As part of this we should allow each component to have different tags (it's globally applied right now from that single value).

### PR DESCRIPTION
**Pull request by gdasson**
Currently, there is no place to specify the image tag for the
following components: kuma dp initContainer,kuma dp container,
kuma-cp and kumactl. Right now the image tag is pulled from
app version in the Chart file and applied to all components.
This fix provides the option to specify different tags for
kuma dp initContainer,kuma dp container,kuma-cp and kumactl

* adds tag to following components: kuma dp initContainer,
  kuma dp container,kuma-cp and kumactl in Helm charts

Fix #3988 (https://github.com/kumahq/kuma/issues/3988)

Signed-off-by: Gaurav Dasson <gaurav.dasson@gmail.com>

### CODE REVIEW

File: deployments/charts/kuma/values.yaml [hunk 1]
```diff
@@ -127,6 +127,8 @@ controlPlane:
     pullPolicy: IfNotPresent
     # -- Kuma CP image repository
     repository: "kuma-cp"
+    # -- Kuma CP Image tag
```
Comment (jakubdyszkiewicz) [inline deployments/charts/kuma/values.yaml:130]: how about
```
    # -- Kuma CP Image tag. When not specified, the value is copied from global.tag
```

**Review by jakubdyszkiewicz — COMMENTED** [review]
Thanks for the contribution!

Please run `make check` to update HELM readme.

File: deployments/charts/kuma/values.yaml (hunk 1 above)
Comment (gdasson) [reply deployments/charts/kuma/values.yaml:130]: make check is giving me some weird errors:
/Users/dasson/bin/protobuf.d/include: warning: directory does not exist.
/Users/dasson/go/pkg/mod/github.com/envoyproxy/protoc-gen-validate@v0.4.1: warning: directory does not exist.
/Users/dasson/go/pkg/mod/github.com/envoyproxy/data-plane-api@: warning: directory does not exist.
/Users/dasson/go/pkg/mod/github.com/cncf/udpa@: warning: directory does not exist.
/Users/dasson/go/pkg/mod/github.com/googleapis/googleapis@: warning: directory does not exist.
validate/validate.proto: File not found.
mesh/v1alpha1/dataplane.proto:12:1: Import "validate/validate.proto" was not found or had errors.
make: *** [protoc/mesh/v1alpha1] Error 1

Tried resolving it, but to no success :(

Comment (gdasson) [reply deployments/charts/kuma/values.yaml]: Done. Fixed.

**Comment by gdasson** [comment]
@jakubdyszkiewicz : I added a Signed off by msg in my commits, but its not liking what I did :(

**Comment by lahabana** [comment]
@gdasson here's some details: https://github.com/kumahq/kuma/pull/4032/checks?check_run_id=5711446574

**Comment by gdasson** [comment]
I am not sure if the steps I did to fix DCO issue are correct. Let me open a new pull request from a new branch.

**Pull request closed without merging**